src/sentry/tasks/groupowner.py

```diff
@@ -20,84 +23,98 @@
 logger = logging.getLogger("tasks.groupowner")
 
 
-def process_suspect_commits(event, **kwargs):
+@instrumented_task(
+    name="sentry.tasks.process_suspect_commits",
+    queue="group_owners.process_suspect_commits",
+    default_retry_delay=5,
+    max_retries=5,
+)
+def process_suspect_commits(event_id, event_platform, event_frames, group_id, project_id, **kwargs):
     metrics.incr("sentry.tasks.process_suspect_commits.start")
-    with metrics.timer("sentry.tasks.process_suspect_commits"):
-        can_process = True
-        # Abbreviation for "workflow-owners-ingestion:group-{}"
-        cache_key = "w-o-i:g-{}".format(event.group_id)
+    lock = locks.get("w-o:{}-d-l".format(group_id), duration=10,)
+    try:
+        with lock.acquire():
+            with metrics.timer("sentry.tasks.process_suspect_commits"):
+                can_process = True
+                # Abbreviation for "workflow-owners-ingestion:group-{}"
+                cache_key = "w-o-i:g-{}".format(group_id)
 
-        if cache.get(cache_key):
-            # Only process once per OWNER_CACHE_LIFE seconds.
-            metrics.incr(
-                "sentry.tasks.process_suspect_commits.skipped", tags={"detail": "too_many_owners"}
-            )
-            can_process = False
-        else:
-            project = Project.objects.get_from_cache(id=event.project_id)
-            owners = GroupOwner.objects.filter(
-                group_id=event.group_id,
-                project=project,
-                organization_id=project.organization_id,
-                type=GroupOwnerType.SUSPECT_COMMIT.value,
-            )
-            owner_count = owners.count()
-            if owner_count >= PREFERRED_GROUP_OWNERS:
-                owners = owners.filter(
-                    date_added__lte=timezone.now() - PREFERRED_GROUP_OWNER_AGE
-                ).order_by("-date_added")
-                if not owners.exists():
+                if cache.get(cache_key):
+                    # Only process once per OWNER_CACHE_LIFE seconds.
                     metrics.incr(
-                        "sentry.tasks.process_suspect_commits.aborted",
-                        tags={"detail": "maxed_owners_none_old"},
+                        "sentry.tasks.process_suspect_commits.skipped",
+                        tags={"detail": "too_many_owners"},
                     )
                     can_process = False
```

Comment:
Nit: If we wanted to, we could just return  here and not indent on `if can_process` ahead.

Reply:
Yeah this is much cleaner. Getting rid of `can_process` altogether.